controllers/chat/sendSignedMessage.js

```diff
@@ -84,6 +86,32 @@ const sendSignedMesage = async (req, res) => {
     if (createdChannel?.channel?.is_channel_blocked)
       return res.status(403).json(responseError('Channel is blocked'));
 
+    // Prevent if user try to chat with blocked user
+    let channelType = createdChannel?.channel?.channel_type;
+    if (channelType === 4 || channelType === 0) {
+      const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(
+        UserBlockedUser,
+        req.userId
+      );
+      let better_channel_member = createdChannel?.channel?.better_channel_member;
+      const blockedIdsSet = new Set(blockedIds);
+      const isBlocked = better_channel_member.some(
+        (member) => member.user_id !== req.userId && blockedIdsSet.has(member.user_id)
+      );
+      if (isBlocked) {
+        return res.status(403).json(responseError('This user does not want to receive messages'));
+      }
+      // if (blockedIds.length > 0) {
+      //   for (const member of better_channel_member) {
+      //     if (member.user_id !== req.userId && blockedIds.includes(member.user_id)) {
+      //       return res
+      //         .status(403)
+      //         .json(responseError('This user does not want to receive messages'));
+      //     }
+      //   }
+      // }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a user is blocked seems correct. However, the code could be optimized by moving the block check into a separate function to improve modularity and readability. Also, consider handling the case where `getBlockedAndBlockerUserId` or `better_channel_member.some` throws an error.

```diff
-    // Prevent if user try to chat with blocked user
-    let channelType = createdChannel?.channel?.channel_type;
-    if (channelType === 4 || channelType === 0) {
-      const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(
-        UserBlockedUser,
-        req.userId
-      );
-      let better_channel_member = createdChannel?.channel?.better_channel_member;
-      const blockedIdsSet = new Set(blockedIds);
-      const isBlocked = better_channel_member.some(
-        (member) => member.user_id !== req.userId && blockedIdsSet.has(member.user_id)
-      );
-      if (isBlocked) {
-        return res.status(403).json(responseError('This user does not want to receive messages'));
-      }
-    }

+    try {
+      await checkIfUserIsBlocked(createdChannel, req.userId);
+    } catch (error) {
+      return res.status(403).json(responseError('This user does not want to receive messages'));
+    }

// Add this function outside of your current function
async function checkIfUserIsBlocked(createdChannel, userId) {
+  let channelType = createdChannel?.channel?.channel_type;
+  if (channelType === 4 || channelType === 0) {
+    const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(
+      UserBlockedUser,
+      userId
+    );
+    let better_channel_member = createdChannel?.channel?.better_channel_member;
+    const blockedIdsSet = new Set(blockedIds);
+    const isBlocked = better_channel_member.some(
+      (member) => member.user_id !== userId && blockedIdsSet.has(member.user_id)
+    );
+    if (isBlocked) {
+      throw new Error('User is blocked');
+    }
+  }
+}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->